pkg/cmd/builder.go

```diff
@@ -243,7 +245,7 @@ func mergeOpenAPIDefinitions(definitionsGetters []openapicommon.GetOpenAPIDefini
 	}
 }
 
-func (b *AdapterBase) defaultOpenAPIConfig() *openapicommon.Config {
+func (b *AdapterBase) defaultOpenAPIConfig(createConfig func(getDefinitions openapicommon.GetOpenAPIDefinitions, defNamer *openapinamer.DefinitionNamer) *openapicommon.Config) *openapicommon.Config {
```

Comment:
I find it quite confusing to have the function passed as an argument when we would expect the function to not take any and just return whatever the default config is.

what do you think about splitting that into 3 functions:
- openAPIConfig(createConfig func(getDefinitions openapicommon.GetOpenAPIDefinitions, defNamer *openapinamer.DefinitionNamer) *openapicommon.Config)
- defaultOpenAPIConfig()
- defaultOpenAPIV3Config()

Reply:
Yes, I should definitely have done that!
I've updated the PR.